### ISSUE TITLE
Accessibility: Duplicated query editors fields IDs

### ISSUE THREAD

**Issue by Elfo404** [grafana/grafana#70761]
Quite a bunch of query editor rely on the query `refId` to generate html ids for input and labels. While this works ok in dashboards, it doesn't necessarily work in Explore or in general in places where multiple set of queries may be shown at the same time as `refId` is only guaranteed to be unique withing a single `QueryRows`.
we should stop relying on `refId` and most likely migrate to `useId` fro, `react`.

**Comment by Elfo404** [comment]
@grafana/observability-metrics @grafana/observability-logs @grafana/grafana-bi-squad could you have a look at this please? 

ref: https://raintank-corp.slack.com/archives/CHKLEJ668/p1684405725051549
